release/utils.py

```diff
@@ -55,12 +52,15 @@ def _is_commit_successful(commit_sha):
         Ensures there is at least one status update so that
         commits whose tests haven't started yet are not valid.
         """
-        statuses = github_api.commit_statuses(commit_sha)
-
-        # Determine if all statuses are success
-        passed_tests = all(status['state'] == 'success' for status in statuses)
-
-        return len(statuses) == NUMBER_OF_TEST_SUITES and passed_tests
+        commit_status = github_api.commit_statuses(commit_sha)
+        try:
+            # Determine if the overall commit has passed all checks
+            if commit_status['state'] == 'success':
```

Comment:
Or you can do a commit_status.get('state') to avoid the KeyError (as long as commit_status is not None, in which case you'll need to test for that first or catch the AttributeError: 'NoneType' object has no attribute 'get').